### PR TITLE
Added infinite room generation

### PR DESCRIPTION
**Pull request by Arcademic**

### CODE REVIEW

File: scripts/Room.cs [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
-		RayCast2D rayLeft = GetNode<RayCast2D>("ray_left");
-		RayCast2D rayRight = GetNode<RayCast2D>("ray_right");
-		RayCast2D rayUp = GetNode<RayCast2D>("ray_up");
-		RayCast2D rayDown = GetNode<RayCast2D>("ray_down");
-		var spaceState = GetWorld2D().DirectSpaceState;
-		var query = PhysicsRayQueryParameters2D.Create(Vector2.Zero, new Vector2(-350, 0));
-    	var result = spaceState.IntersectRay(query);
-
-		GD.Print(result);
+		BodyEntered += OnBodyEntered;
```
Comment (tairasu) [inline scripts/Room.cs:11]: clever!

File: scripts/Level1.cs [hunk 2]
```diff
@@ -7,43 +6,55 @@ public partial class Level1 : Node
 	PackedScene[] rooms;
 	
 	Node2D startingRoom;
```
Comment (tairasu) [inline scripts/Level1.cs]: no need to keep that!

File: scripts/Level1.cs [hunk 3]
```diff
@@ -7,43 +6,55 @@ public partial class Level1 : Node
...
 		var roomScene = rooms[GD.Randi() % rooms.Length];
-		Node2D room = (Node2D)roomScene.Instantiate();
 
-		float rotation = GD.Randi() % 4 * (Mathf.Pi / 2);
-		room.Rotation = rotation;
-
-		
-		room.Position = currentRoom.Position + 550 * Vector2.Left;
-		
+		for (int i = -1; i <= 1; i++) {
```
Comment (tairasu) [inline scripts/Level1.cs:32]: clever!

**Review by tairasu — CHANGES_REQUESTED** [review]

File: scripts/Level1.cs (hunk 3 above)
Comment (Arcademic) [reply scripts/Level1.cs:32]: ty

**Review by tairasu — APPROVED** [review]

**Pull request merged**